test/functional/api/extmark_spec.lua

```diff
@@ -1606,3 +1606,188 @@ describe('Extmarks buffer api with many marks', function()
     eq({}, get_marks(ns2))
   end)
 end)
+
+describe('API/win_extmark', function()
+  local screen
+  local marks, line1, line2
+  local ns
+
+  before_each(function()
+    -- Initialize some namespaces and insert text into a buffer
+    marks = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12}
+
+    line1 = "non ui-watched line"
+    line2 = "ui-watched line"
+
+    clear()
+
+    insert(line1)
+    feed("o<esc>")
+    insert(line2)
+    ns = request('nvim_create_namespace', "extmark-ui")
+  end)
+
+  it('sends and only sends ui-watched marks to ui', function()
+    screen = Screen.new(20, 4)
+    screen:attach()
+    -- should send this
+    set_extmark(ns, marks[1], 1, 0, {
+      ui_watched = true
+    })
+    -- should not send this
+    set_extmark(ns, marks[2], 0, 0, {
+      ui_watched = false
+    })
+    -- screen not changed, use sleep to drive the event loop
+    screen:sleep(1)
+    screen:expect({
+      grid = [[
+      non ui-watched line |
+      ui-watched lin^e     |
+      ~                   |
+                          |
+    ]],
+      extmarks = {
+        [2] = {
+          -- positioned at the end of the 2nd line
+          { {id = 1000}, 1, 1, 1, 16 },
+        }
+      },
+      unchanged = true
```

Comment:
the screen is not unchanged.

Reply:
without that:
```
warning: Screen test succeeded immediately. Try to avoid this unless the
purpose of the test really requires it.
If necessary, silence this warning with 'unchanged' argument of screen:expect.
```